maestrowf/datastructures/core/executiongraph.py

```diff
@@ -642,22 +663,30 @@ def execute_ready_steps(self):
                 # as the number of dependencies the step has, it's ready to
                 # be executed. Add it to the map.
                 if num_finished == len(record.step.run["depends"]):
-                    logger.debug("All dependencies completed. Staging.")
-                    ready_steps[key] = record
+                    if key not in self.ready_steps:
+                        logger.debug("All dependencies completed. Staging.")
+                        self.ready_steps.append(record)
+                    else:
+                        logger.debug("Already staged. Passing.")
+                        continue
 
         # We now have a collection of ready steps. Execute.
-        for key, record in ready_steps.items():
-            logger.info("Executing -- '%s'\nScript path = %s", key,
-                        record.script)
-            logger.debug(
-                "Attempting to execute '%s' -- Current state is %s.",
-                record.name, record.status
-            )
-            self._execute_record(key, record)
-            logger.debug(
-                "After execution of '%s' -- New state is %s.",
-                record.name, record.status
-            )
+        # If we don't have a submission limit, go ahead and submit all.
+        if not self._submission_throttle:
```

Comment:
Doesn't this test if `self._submission_throttle` is not `None`? Currently, `self._submission_throttle` can't be `None` right? Since the default is set to `0` on `__init__`.

Reply:
This works because the value of 0 is considered to be a False value.